tests/lax_test.py

```diff
@@ -3365,87 +3365,119 @@ def f(x):
-          'negative', 'absolute', 'sign'],
+    dict(testcase_name=f"_{dtype.__name__}", dtype=dtype)
+    for dtype in jtu.dtypes.supported([np.float32, np.float64, np.complex64, np.complex128]))
+  def testMPMathUtils(self, dtype):
+    mpmath = jtu.pytest_importorskip('mpmath', '1.3')
+
+    prec = {np.float32: 24, np.float64: 53, np.complex64: 24, np.complex128: 53}[dtype]
+    is_complex = dtype().dtype.kind == 'c'
+
+    def func(x):
+      assert isinstance(x, mpmath.ctx_mp.mpnumeric)
+      assert x.context.prec == prec
+      assert isinstance(x, x.context.mpc if is_complex else x.context.mpf)
+      return x
+
+    ufunc = jtu.vectorize_with_mpmath(func, mpmath=mpmath)
+
+    with jtu.ignore_warning(category=RuntimeWarning, message="(overflow|invalid value|divide by zero) encountered in.*"):
+      if is_complex:
+        arr = jtu.complex_plane_sample(dtype=dtype, size_re=11)
+      else:
+        cdtype = getattr(np, ufunc.map_float_to_complex[dtype.__name__])
+        arr = jtu.complex_plane_sample(dtype=cdtype, size_re=11, size_im=0)[1:2].real
+
+    arr2 = ufunc.mptonp(ufunc.nptomp(arr))
+    with jtu.ignore_warning(category=RuntimeWarning, message="(overflow|invalid value|divide by zero) encountered in.*"):
+      self.assertAllClose(arr, arr2, atol=0, rtol=0)
+
+    arr3 = ufunc(arr)
+    with jtu.ignore_warning(category=RuntimeWarning, message="(overflow|invalid value|divide by zero) encountered in.*"):
+      self.assertAllClose(arr, arr3, atol=0, rtol=0)
+
+  _functions_on_complex_plane = [
+    'arccos', 'arccosh', 'arcsin', 'arcsinh',
+    'arctan', 'arctanh', 'conjugate', 'cos',
+    'cosh', 'exp', 'exp2', 'expm1', 'log',
+    'log10', 'log1p', 'sin', 'sinh', 'sqrt',
+    'square', 'tan', 'tanh', 'sinc', 'positive',
+    'negative', 'absolute', 'sign'
+  ]
+
+  @parameterized.named_parameters(
+    dict(testcase_name=f"_{name}_{dtype.__name__}", name=name, dtype=dtype)
+    for name, dtype in itertools.product(
+        _functions_on_complex_plane,
         jtu.dtypes.supported([np.complex64, np.complex128]),
-        ['success', 'failure'],
     ))
   @jtu.skip_on_devices("tpu")
-  def testOnComplexPlane(self, name, dtype, kind):
-    all_regions = ['q1', 'q2', 'q3', 'q4', 'pos', 'neg', 'posj', 'negj', 'ninf', 'pinf', 'ninfj', 'pinfj', 'zero']
+  def testSuccessOnComplexPlane(self, name, dtype):
+    self._testOnComplexPlaneWorker(name, dtype, 'success')
+
+  @parameterized.named_parameters(
+    dict(testcase_name=f"_{name}_{dtype.__name__}", name=name, dtype=dtype)
+    for name, dtype in itertools.product(
+        _functions_on_complex_plane,
+        jtu.dtypes.supported([np.complex64, np.complex128]),
+    ))
+  @jtu.skip_on_devices("tpu")
+  def testFailureOnComplexPlane(self, name, dtype):
+    self._testOnComplexPlaneWorker(name, dtype, 'failure')
+
+  def _testOnComplexPlaneWorker(self, name, dtype, kind):
+
+    mpmath = jtu.pytest_importorskip('mpmath', '1.3')
```

Comment:
Let's make `mpmath` a required test-time dependency to remove the need for this `importorskip` logic.

Reply:
Done.

Reply:
This going to be a problem for `[build with 3.9 (py 3.9 on ubuntu-20.04-16core, x64=1)](https://github.com/google/jax/actions/runs/8378976633/job/22944817602?pr=20144#logs)` that does not have mpmath installed.

Reply:
Resolved by adding mpmath to build/test-requirements.txt